### DISCUSSION TITLE
Is there a way to get whole url?

### DISCUSSION THREAD

**Discussion by monoald** [honojs/hono#1615] [Q&A]
I'm writing a social media auth library for hono. When asking for token to Facebook they send the token on the redirect uri like this:

`endpoint?#access_token=token`

Due to the hash I can't use `c.req.query('access'token')` it will return `undefined`, with `c.req.url` I only get `endpoint?` I can't get the info after the question mark.

Is there a way to get whole url?

**Comment by yusukebe** [comment]
Hi.

You can't do this without a front-end trick. The server app can't recognize the hash values in the URL. See: https://stackoverflow.com/a/5536699

**Reply by monoald** [reply]
Thank you very much for your time and the insight, don't know if this is a proper solution but it works.

```
return c.html(html`
  <html>
    <head>
        <script>
          const url = window.location.href.replace('#', '')
          window.location.href = url
        </script>
    </head>
  </html>
  `
)
```

This way I can "clean" the url and retrieve the data with `c.req.query()`.
